src/clr-angular/data/datagrid/datagrid.ts

```diff
@@ -91,7 +99,30 @@ export class ClrDatagrid implements AfterContentInit, AfterViewInit, OnDestroy {
     /**
      * We grab the smart iterator from projected content
      */
-    @ContentChild(ClrDatagridItems) public iterator: ClrDatagridItems;
+    // TODO - Add correct type - rebase with: https://github.com/vmware/clarity/pull/2347
+    @ContentChild(ClrDatagridItems) public iterator: ClrDatagridItems<RowContext<any>>;
+
+    /**
+     * Set the state of the pinned first column
+     * default is false.
+     */
+
+    public _pinnedColumn: boolean = false;
```

Comment:
Matt and I have talked about this a little bit, while I think it would be nice to have the the pinned columns pieces separated from here, they are pretty minor and inert. What do you think @youdz ?

Reply:
This I forgot to remove from before starting the render refactor. Will remove it for now. 

Pulling the pinned column demos out is easy - they are just a simple html/ts demo. Either way is fine with me.